pyproject.toml

```diff
@@ -31,6 +31,7 @@ classifiers = [
 requires-python = ">=3.10"
 dependencies = [
   "numpy >= 1.20",
+  "psutil",
```

Comment:
I am not comfortable adding this as a new required dependency for all of Scipp. Can we do without it?

Reply:
We could fallback to using `os.process_cpu_count` https://docs.python.org/3/library/os.html#os.process_cpu_count if psutil is not installed.

The drawback is that `process_cpu_cores` returns the number of logical cores but for best performance we want the number of physical cores.

Maybe falling back to `os.process_cpu_count//2` is a good solution?

Reply:
How about using the value of the `workers` argument, then the user can decide?

Reply:
Yeah maybe that's just easiest. I was thinking it's good to have an option to "use all cores", but probably it's not necessary, especially since we're making the multiprocessing opt-in anyway.